tests/integration_tests/test_cli.py

```diff
@@ -298,164 +298,150 @@ def test_that_setenv_sets_environment_variables_in_jobs(setenv_config):
             dedent(
                 """
-                    #!/bin/bash
-                    ekho helo wordl
-                    STOP_ON_FAIL=False
+                    INTERNAL True
+                    SCRIPT failing_script.py
                 """
             ),
-            True,
-        ),
-        (
+            "py",
             dedent(
                 """
-                   STOP_ON_FAIL False
-                   INTERNAL False
-                   EXECUTABLE failing_script.sh
-                """
-            ),
-            "failing_script.sh",
-            dedent(
+                    from ert import ErtScript
+                    class AScript(ErtScript):
+
+                        def run(self):
+                            assert False, "failure"
                 """
-                   #!/bin/bash
-                   ekho helo wordl
-                   STOP_ON_FAIL=TRUE
-               """
             ),
             False,
+            id="internal_python_script__stop_on_fail_disabled_by_default",
         ),
-        (
+        pytest.param(
             dedent(
                 """
-                    INTERNAL False
-                    EXECUTABLE failing_script_w_stop.sh
+                    INTERNAL True
+                    SCRIPT failing_script.py
+                    STOP_ON_FAIL False
                 """
             ),
-            "failing_script_w_stop.sh",
+            "py",
             dedent(
                 """
-                    #!/bin/bash
-                    ekho helo wordl
-                    STOP_ON_FAIL=True
+                    from ert import ErtScript
+                    class AScript(ErtScript):
+
+                        def run(self):
+                            assert False, "failure"
                 """
             ),
-            True,
+            False,
+            id="internal_python_script__stop_on_failed_disabled",
         ),
-        (
+        pytest.param(
             dedent(
                 """
                     INTERNAL True
-                    SCRIPT failing_ert_script.py
+                    SCRIPT failing_script.py
+                    STOP_ON_FAIL True
                 """
             ),
-            "failing_ert_script.py",
-            """
-from ert import ErtScript
-class AScript(ErtScript):
-    stop_on_fail = True
-
-    def run(self):
-        assert False, "failure"
-""",
-            True,
-        ),
-        (
+            "py",
             dedent(
                 """
-                    INTERNAL True
-                    SCRIPT failing_ert_script.py
-                    STOP_ON_FAIL False
+                    from ert import ErtScript
+                    class AScript(ErtScript):
+
+                        def run(self):
+                            assert False, "failure"
+
                 """
             ),
-            "failing_ert_script.py",
-            """
-from ert import ErtScript
-class AScript(ErtScript):
-    stop_on_fail = True
```

Comment:
Note that we do want to keep this `stop_on_fail = True` within `ErtScript`s functionality.